Adding MIT License (#104)

VaishnaviHire opened the pull request:
This is a re-make of #98 , that was prematurely approved (sorry, my bad).

People have still some concerns that we need to resolve before merging the license. It can be an easy discussion, but IANAL, so better play safe and listen to some more voices.

The questions actually are:

* What is the implication to have `Copyright (c) 2025 Meta Llama` wrt to future work and when moving potentially to a foundation later ? Wouldn't it be bettet to use a neutral term like `Copyright (c) 2025 The Llama Stack community` ?
* There are still repos in the GitHub `llamastack` org that uses an APL license, eg. [llama-stack-client-typescript/LICENSE](https://github.com/llamastack/llama-stack-client-typescript/blob/generated/LICENSE), but I have the feeling all repos in that org should leverage the same license.
* Also, other projects that are using the MIT license, have a different copyright holder line, like `Copyright (c) Meta Platforms, Inc. and affiliates` in [llamastack/llama-stack-client-python](https://github.com/llamastack/llama-stack-client-python/blob/fe7130c105885a97cbc67410050a45f3de2f2db5/LICENSE)

My proposal would be to settle on a single license file with the same content and update all repositories atomically.

Approved-by: leseb

Approved-by: VaishnaviHire

Approved-by: nathan-weinberg
(cherry picked from commit 2b48cb807144eafda84efc135ab4135ede69529b)

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Added an open source license (MIT License) to the project.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rhuss commented:
Looks great, thanks! I wonder whether we should add some unit test (although I don't know whether such a test would then just test the golang core and controller runtime core functionalities, so might not be worth the hassle)

rhuss committed:
Adding MIT License (#104)

This is a re-make of #98 , that was prematurely approved (sorry, my bad).

People have still some concerns that we need to resolve before merging the license.
It can be an easy discussion, but IANAL, so better play safe and listen to some more voices.

The questions actually are:

* What is the implication to have `Copyright (c) 2025 Meta Llama` wrt to future work and when moving potentially to a foundation later ? Wouldn't it be bettet to use a neutral term like `Copyright (c) 2025 The Llama Stack community` ?
* There are still repos in the GitHub `llamastack` org that uses an APL license, eg. [llama-stack-client-typescript/LICENSE](https://github.com/llamastack/llama-stack-client-typescript/blob/generated/LICENSE), but I have the feeling all repos in that org should leverage the same license.
* Also, other projects that are using the MIT license, have a different copyright holder line, like `Copyright (c) Meta Platforms, Inc. and affiliates` in [llamastack/llama-stack-client-python](https://github.com/llamastack/llama-stack-client-python/blob/fe7130c105885a97cbc67410050a45f3de2f2db5/LICENSE)

My proposal would be to settle on a single license file with the same content and update all repositories atomically.

Approved-by: leseb

Approved-by: VaishnaviHire

Approved-by: nathan-weinberg
(cherry picked from commit 2b48cb807144eafda84efc135ab4135ede69529b)
